config.py

```diff
@@ -6,26 +6,59 @@
 config_bp = Blueprint('config', __name__)
 
 
-@config_bp.route('/next_year', methods=['POST'])
+# This method allows the admin to create a new academic year.
+@config_bp.route('/new_year', methods=['POST'])
 @login_required
 @check_access_level(Role.ADMIN)
-def next_year():
+def new_year():
     last_year = Configuration.query.order_by(Configuration.year.desc()).first()
     new_year = last_year.year + 1
 
     try:
         # Check whether the following year already exists in the database
         existing_config = Configuration.query.filter_by(year=new_year).first()
         if existing_config is not None:
-            return make_response("The year already exists", 500)
+            flash("The year already exists", "error")
+            return redirect(url_for("index"))
 
         config = Configuration(year=new_year)
         db.session.add(config)
         db.session.commit()
-        Configuration.update_current_year(config.id)
 
     except Exception as e:
         db.session.rollback()
-        raise e
+        flash("An error occurred while processing your request. Please try again.", "error")
```

Comment:
Maybe we can log the error or at least give an insight in the feedback of what's gone wrong. There is almost no chance that "trying again" will work.